frame/staking/src/pallet/impls.rs

```diff
@@ -1308,12 +1296,9 @@ impl<T: Config> SortedListProvider<T::AccountId> for UseNominatorsMap<T> {
 		Ok(())
 	}
 	fn clear(maybe_count: Option<u32>) -> u32 {
+		let count_before = Nominators::<T>::count();
 		Nominators::<T>::remove_all(maybe_count);
-		if let Some(count) = maybe_count {
-			CounterForNominators::<T>::mutate(|noms| *noms - count);
-			count
-		} else {
-			CounterForNominators::<T>::take()
-		}
+		let count_after = Nominators::<T>::count();
+		count_before.saturating_sub(count_after)
```

Comment:
I dont think this returns the same value as before?

Reply:
```
	/// Remove `maybe_count` number of items from the list. Returns the number of items actually
	/// removed. WARNING: removes all items if `maybe_count` is `None`, which should never be done
	/// in production settings because it can lead to an unbounded amount of storage accesses.
	```

Reply:
It should but I am not 100% certain.

Reply:
sidenote: actually the implementation was wrong before, it should be fixed by https://github.com/paritytech/substrate/pull/10231/files

Reply:
Ah...okay